CryptoTechReminderSystem.Test/SlackGatewayTests.cs

```diff
@@ -15,91 +17,145 @@ public class SlackGatewayTests
+                var message = new Message()
+                {
+                    Channel = "U0112WTW",
+                    Text = text
+                };
+            
+                _slackGateway.Send(message);
+            
+                var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+                
+                JObject.Parse(receivedRequest.RequestBody)["text"].ToString().Should().Be(text);
+            }
+        }
+
+        [TestFixture]
+        public class CanRetrieveDevelopers
         {
-            var text = "Please make sure your timesheet is submitted by 13:30 on Friday.";
-            var message = new Message()
+            private FluentSimulator _fluentSimulator;
+            private SlackGateway _slackGateway;
+            private IList<SlackDeveloper> _response;
+
+            [SetUp]
+            public void SetUp()
             {
-                Channel = "U0112WTW",
-                Text = text
-            };
+                _fluentSimulator = new FluentSimulator(Address);
+                _slackGateway = new SlackGateway(Address, Token);
             
-            _slackGateway.Send(message);
+                var json = File.ReadAllText(
+                    Path.Combine(
+                        AppDomain.CurrentDomain.BaseDirectory,
+                        "../../../SlackUsersExampleResponse.json"
+                    )
+                );
+                
+                _fluentSimulator.Get("/api/users.list").Responds(json);
+            
+                _fluentSimulator.Start();
+
+                _response = _slackGateway.RetrieveDevelopers();
+            }
             
-            var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+            [TearDown]
+            public void TearDown()
+            {
+                _fluentSimulator.Stop();
+            }
+            
+            [Test]
+            public void CanSendAGetUsersRequest()
+            {
+                var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+                
+                receivedRequest.Url.Should().Be(Address + "api/users.list");
+            }
+            
+            [Test]
+            public void CanSendAGetUsersRequestWithAToken()
+            {
+                var receivedRequest = _fluentSimulator.ReceivedRequests.First();
                 
-            DeserializeObject<PostMessageRequest>(receivedRequest.RequestBody).Text.Should().Be(text);
+                receivedRequest.Headers["Authorization"].Should().Be("Bearer " + Token);
+            }
+        
+            [Test]
+            public void CanGetAListOfSlackDevelopers()
+            {
+                _response.Should().BeOfType<List<SlackDeveloper>>();
+                _response.Should().HaveCount(6);
+            }
+        
+            [Test]
+            public void CanGetIdOfSlackDeveloper()
+            {
+                _response.First().Id.Should().Be("W0123CHAN");
```

Comment:
It would be good to have an additional assertion on the email to which this channel Id links to, this would have caught the previous issue of nor returning the email from the gateway

Reply:
Maybe another test instead of another assertion is a good idea?

Reply:
Yes, I've done this in https://github.com/chubberlisk/crypto_tech/pull/12/commits/5f84738fbd9c01f8099cba3b26f24fa62fb1fa7d. See more in https://github.com/chubberlisk/crypto_tech/pull/12#discussion_r273994444.